docs/aws/audit/opensearchmonitoring/rules/opensearch_https_required_remediation.mdx

```diff
@@ -1,6 +1,220 @@
+   - In the search bar, type "OpenSearch Service" and select it from the list of services.
+
+2. **Select the Domain:**
+   - In the OpenSearch Service dashboard, you will see a list of your OpenSearch domains.
+   - Click on the domain name for which you want to enforce HTTPS.
+
+3. **Edit Domain Configuration:**
+   - In the domain details page, look for the "Actions" button and click on it.
+   - Select "Edit domain" from the dropdown menu.
+
+4. **Enable HTTPS:**
+   - In the "Network configuration" section, find the "Endpoint" settings.
+   - Ensure that the "Require HTTPS" option is checked.
+   - Save the changes by clicking the "Save changes" button at the bottom of the page.
+
+By following these steps, you can ensure that HTTPS is enforced for your OpenSearch Service domains, enhancing the security of your data in transit.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the misconfiguration where HTTPS should be enforced for OpenSearch Service Domains in AWS using the AWS CLI, follow these steps:
+
+1. **Describe the OpenSearch Domain Configuration:**
+   First, you need to get the current configuration of your OpenSearch domain to understand its settings.
+   ```sh
+   aws opensearch describe-domain --domain-name <your-domain-name>
+   ```
+
+2. **Modify the Domain Configuration to Enforce HTTPS:**
+   Update the domain configuration to enforce HTTPS by setting the `EnforceHTTPS` parameter to `true`.
+   ```sh
+   aws opensearch update-domain-config --domain-name <your-domain-name> --domain-endpoint-options EnforceHTTPS=true
+   ```
+
+3. **Verify the Configuration Update:**
+   After updating the domain configuration, verify that the changes have been applied correctly.
+   ```sh
+   aws opensearch describe-domain --domain-name <your-domain-name>
+   ```
+
+4. **Monitor the Domain Status:**
+   Ensure that the domain status is active and that the changes have been successfully implemented.
+   ```sh
+   aws opensearch describe-domain --domain-name <your-domain-name> --query 'DomainStatus.Processing'
+   ```
+
+Replace `<your-domain-name>` with the actual name of your OpenSearch domain. These steps will help ensure that HTTPS is enforced for your OpenSearch Service Domains using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration where HTTPS should be enforced for OpenSearch Service Domains in AWS using Python scripts, you can follow these steps:
+
+### Step 1: Install Boto3
+Ensure you have the Boto3 library installed, which is the AWS SDK for Python. You can install it using pip if you haven't already:
+
+```bash
+pip install boto3
+```
+
+### Step 2: Set Up AWS Credentials
+Make sure your AWS credentials are configured. You can set them up using the AWS CLI or by setting environment variables:
+
+```bash
+aws configure
+```
+
+Or set environment variables:
+
+```bash
+export AWS_ACCESS_KEY_ID='your_access_key_id'
+export AWS_SECRET_ACCESS_KEY='your_secret_access_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>